libs/langchain/langchain/schema/runnable.py

```diff
@@ -751,32 +777,21 @@ def __ror__(
             )
 
     def invoke(self, input: Input, config: Optional[RunnableConfig] = None) -> Output:
-        from langchain.callbacks.manager import CallbackManager
-
         # setup callbacks
-        config = config or {}
-        callback_manager = CallbackManager.configure(
-            inheritable_callbacks=config.get("callbacks"),
-            local_callbacks=None,
-            verbose=False,
-            inheritable_tags=config.get("tags"),
-            local_tags=None,
-            inheritable_metadata=config.get("metadata"),
-            local_metadata=None,
-        )
+        config = config or _empty_config()
```

Comment:
Do we need to ensure here that locals is not none

Reply:
_empty_config sets to {}

Reply:
But non empty config doesn't?

Reply:
like if `config` is not None? given that RunnableConfig has _locals on it it *should* have something for _locals, but i'm guessing that's not actually enforced anywhere? so yea maybe worth checking

Reply:
Yea runnableconfig is just a typed dict, so no defaults apply etc

Reply:
We already have a get config list helper method, can make a get config method helper method that takes care to assign these defaults?

Reply:
darn, i thought TypedDict might at least throw an error if the specified keys were missing